static/app/components/events/rrwebReplayer/baseRRWebReplayer.tsx

```diff
@@ -0,0 +1,333 @@
+import {useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+import RRWebPlayer from 'rrweb-player';
+
+import space from 'sentry/styles/space';
+
+type RRWebEvents = ConstructorParameters<typeof RRWebPlayer>[0]['props']['events'];
+
+interface Props {
+  className?: string;
+  events?: RRWebEvents;
+}
+
+const RRWebReplayerComponent = ({events, className}: Props) => {
+  const [playerEl, setPlayerEl] = useState<HTMLDivElement | null>(null);
+  // const [events, setEvents] = useState<RRWebEvents>();
+
+  // const loadEvents = async () => {
+  // try {
+  // const resp = await fetch(url);
+  // const data = await resp.json();
+
+  // setEvents(data.events);
+  // } catch (err) {
+  // Sentry.captureException(err);
+  // }
+  // };
+
+  // useEffect(() => void loadEvents(), [url]);
+
+  const initPlayer = () => {
+    if (events === undefined) {
+      return;
+    }
+
+    if (playerEl === null) {
+      return;
+    }
+
+    // eslint-disable-next-line no-new
+    new RRWebPlayer({
+      target: playerEl,
+      props: {events, autoPlay: false},
+    });
+  };
+
+  useEffect(() => void initPlayer(), [events, playerEl]);
```

Comment:
There's no way to update an existing RRWebPlayer with new events right? YOu have to re-inialize it?

Reply:
I believe that's correct

Reply:
Hmm we should probably use `useRef` for playerEl.